CMakeLists.txt

```diff
@@ -10,18 +10,21 @@ cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
 project(GQCP VERSION 0.0.1 LANGUAGES CXX)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)  # make CMake look into the ./cmake/ folder for configuration files
 
+# Add the versioning as a separate library. Whenever the version changes, only the version.cpp source file will be recompiled 
+# and only gqcp needs to be relinked. This way, gqcp does not need to be recompiled if the version should be modified.
 # Add configuration file containing version of current library.
 configure_file(${CMAKE_SOURCE_DIR}/cmake/version.hpp.in version/version.hpp)
+configure_file(${CMAKE_SOURCE_DIR}/cmake/version.cpp.in version/version.cpp)
 # Compile the version as its own library where, if needed, other targets can be linked to.
 # Common practice: this to avoid unneccessary recompiling whenever version details change.
```

Comment:
This documentation seems superfluous, given the documentation that you added a couple of lines above it.